activities/models/emoji.py

```diff
@@ -146,15 +146,18 @@ def load_locals(cls) -> dict[str, "Emoji"]:
 
     @classmethod
     @cached(cache=TTLCache(maxsize=1000, ttl=60))
-    def get_by_domain(cls, shortcode, domain: Domain | None) -> "Emoji":
+    def get_by_domain(cls, shortcode, domain: Domain | None) -> "Emoji | None":
```

Comment:
Can we have the return type here as `"Emoji" | None` please?

Reply:
That is what I did initially, but I got an error. Let me try it again and capture the error

Reply:
Yeah, it explodes during the docker build, even though pre-commit accepts it. Kinda strange:

```
 => ERROR [9/9] RUN TAKAHE_DATABASE_SERVER="postgres://x@example.com/x" TAKAHE_SECRET_KEY="takahe" python3 manage.py collectstatic --noinput                                                                                                                                                           2.2s
------                                                                                                                                                                                                                                                                                                      
 > [9/9] RUN TAKAHE_DATABASE_SERVER="postgres://x@example.com/x" TAKAHE_SECRET_KEY="takahe" python3 manage.py collectstatic --noinput:                                                                                                                                                                      
#12 1.936 Traceback (most recent call last):                                                                                                                                                                                                                                                                
#12 1.936   File "/takahe/manage.py", line 38, in <module>                                                                                                                                                                                                                                                  
#12 1.937     main()                                                                                                                                                                                                                                                                                        
#12 1.937   File "/takahe/manage.py", line 34, in main                                                                                                                                                                                                                                                      
#12 1.937     execute_from_command_line(sys.argv)
#12 1.937   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
#12 1.937     utility.execute()
#12 1.937   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 420, in execute
#12 1.938     django.setup()
#12 1.938   File "/usr/local/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
#12 1.938     apps.populate(settings.INSTALLED_APPS)
#12 1.938   File "/usr/local/lib/python3.11/site-packages/django/apps/registry.py", line 116, in populate
#12 1.938     app_config.import_models()
#12 1.938   File "/usr/local/lib/python3.11/site-packages/django/apps/config.py", line 269, in import_models
#12 1.938     self.models_module = import_module(models_module_name)
#12 1.938                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 1.940   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
#12 1.940     return _bootstrap._gcd_import(name[level:], package, level)
#12 1.941            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 1.942   File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
#12 1.942   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
#12 1.942   File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
#12 1.942   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
#12 1.942   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
#12 1.942   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
#12 1.942   File "/takahe/activities/models/__init__.py", line 1, in <module>
#12 1.942     from .emoji import Emoji, EmojiStates  # noqa
#12 1.942     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 1.942   File "/takahe/activities/models/emoji.py", line 83, in <module>
#12 1.942     class Emoji(StatorModel):
#12 1.942   File "/takahe/activities/models/emoji.py", line 149, in Emoji
#12 1.943     def get_by_domain(cls, shortcode, domain: Domain | None) -> "Emoji" | None:
#12 1.943                                                                 ~~~~~~~~^~~~~~
#12 1.944 TypeError: unsupported operand type(s) for |: 'str' and 'NoneType'
```

Reply:
The spec doesn't really discuss this, so I think its either "do it inside the quotes" or "use `Optional`"

Reply:
How strange! I'll take it as is then.